pkg/secrets/secrets.go

```diff
@@ -47,11 +47,34 @@ func Init(command string, arguments []string, timeout int, maxSize int) {
 
 type walkerCallback func(string) (string, error)
 
+// Viper support setting chunk of configuration through env variable using
+// Yaml/json. Sadly those are loaded on the fly when querying the
+// configuration, the underlying type remain a string. In order to support
+// nested `ENC` in JSON payload we try to unmarshal each string. This is costly
+// but only done once when loading the configuration when the agent starts.
+func handleString(str string, callback walkerCallback) (interface{}, error) {
+	var data interface{}
+	err := yaml.Unmarshal([]byte(str), &data)
+
+	if err == nil {
+		switch v := data.(type) {
```

Comment:
Can there be cases where there is deeper nesting?

Reply:
Technically yes and we would handle them. `WalkHash` and `WalkSlice` would load them. But in real life there is no use case for it.

We trying to solve this case:
```bash
DD_ADDITIONAL_ENDPOINTS='{"https://test": ["ENC[my_api_key]"]}' datadog-agent run
```